syndeo-template/src/test/java/com/google/cloud/syndeo/it/PubsubDlqIT.java

```diff
@@ -0,0 +1,335 @@
+  protected static final CredentialsProvider CREDENTIALS_PROVIDER =
+      FixedCredentialsProvider.create(CREDENTIALS);
+
+  private static final Integer MESSAGE_COUNT = 10;
+  private static final Long ONE_MINUTE_MILLIS = 60 * 1000L;
+
+  @Rule public final TestName testName = new TestName();
+
+  @Rule public final TestPipeline mainPipeline = TestPipeline.create();
+
+  // public static final String JSON_SCHEMA = "{\r\n        \"$id\":
+  // \"https:\/\/example.com\/person.schema.json\",\r\n        \"$schema\":
+  // \"https:\/\/json-schema.org\/draft-07\/schema\",\r\n        \"title\": \"pubsub data
+  // schema\",\r\n        \"type\": \"object\",\r\n        \"properties\": {\r\n          \"name\":
+  // {\r\n            \"type\": \"string\"\r\n          }\r\n        }\r\n      }";
+
+  private TopicName pubsubSourceTopic = null;
+  private SubscriptionName pubsubSourceSubscription = null;
+  private TopicName pubsubSinkTopic = null;
+  private SubscriptionName pubsubSinkSubscription = null;
+  private TopicName pubsubDlqTopic = null;
+  private SubscriptionName pubsubDlqSubscription = null;
+  private PubsubResourceManager pubsubResourceManager = null;
+
+  @Before
+  public void setUpPubSub() throws IOException {
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(TEST_ID, PROJECT)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+    RESOURCE_MANAGERS.add(pubsubResourceManager);
+
+    pubsubSourceTopic =
+        pubsubResourceManager.createTopic("source-" + TEST_ID + testName.getMethodName());
+    LOG.info("Successfully created topic {}", pubsubSourceTopic);
+    pubsubSourceSubscription =
+        pubsubResourceManager.createSubscription(
+            pubsubSourceTopic, "sub-" + pubsubSourceTopic.getTopic());
+    LOG.info("Successfully created subscription {}", pubsubSourceSubscription);
+
+    pubsubSinkTopic =
+        pubsubResourceManager.createTopic("sink-" + TEST_ID + testName.getMethodName());
+    LOG.info("Successfully created topic {}", pubsubSinkTopic);
+    pubsubSinkSubscription =
+        pubsubResourceManager.createSubscription(
+            pubsubSinkTopic, "sub-" + pubsubSinkTopic.getTopic());
+    LOG.info("Successfully created subscription {}", pubsubSinkSubscription);
+
+    pubsubDlqTopic = pubsubResourceManager.createTopic("dlq-" + TEST_ID + testName.getMethodName());
+    LOG.info("Successfully created topic {}", pubsubDlqTopic);
+    pubsubDlqSubscription =
+        pubsubResourceManager.createSubscription(
+            pubsubDlqTopic, "sub-" + pubsubDlqTopic.getTopic());
+    LOG.info("Successfully created subscription {}", pubsubDlqSubscription);
+  }
+
+  @After
+  public void cleanUp() {
+    ResourceManagerUtils.cleanResources(RESOURCE_MANAGERS.toArray(new ResourceManager[0]));
+    pubsubSourceTopic = null;
+    pubsubSinkTopic = null;
+    pubsubDlqTopic = null;
+    pubsubResourceManager = null;
+  }
+
+  @Test
```

Comment:
Eventually (not this PR), we should also add tests to stress the message size and see whats the upper limit for publishing messages to pubsub. And then we should use that value to revise our 10MB limit in the transform.

Reply:
Yeah, sounds good.